.vitepress/config.mjs

```diff
@@ -0,0 +1,179 @@
+import { defineConfig } from 'vitepress'
+import tabsPlugin from './plugins/tabs-plugin';
+import istioSidebar from '../docs/external-content/istio/docs/user/_sidebar';
+import apiGatewaySidebar from '../docs/external-content/api-gateway/docs/user/_sidebar';
+import sapBtpOperatorSidebar from '../docs/external-content/btp-manager/docs/user/_sidebar';
+import applicationConnectorSidebar from '../docs/external-content/application-connector-manager/docs/user/_sidebar';
+import cloudManagerSidebar from '../docs/external-content/cloud-manager/docs/user/_sidebar';
+import dockerRegistrySidebar from '../docs/external-content/docker-registry/docs/user/_sidebar';
+import eventingSidebar from '../docs/external-content/eventing-manager/docs/user/_sidebar';
+import kedaSidebar from '../docs/external-content/keda-manager/docs/user/_sidebar';
+import natsSidebar from '../docs/external-content/nats-manager/docs/user/_sidebar';
+import serverlessSidebar from '../docs/external-content/serverless-manager/docs/user/_sidebar';
+import telemetrySidebar from '../docs/external-content/telemetry-manager/docs/user/_sidebar';
+import cliSidebar from '../docs/external-content/cli/docs/user/_sidebar';
+import busolaSidebar from '../docs/external-content/busola/docs/user/_sidebar';
+
+export function getSearchConfig() {
+  return {
+      provider: 'local',
+      detailedView: true,
+      options: {
+        detailedView: true,
+        miniSearch: {
+          /**
+           * @type {Pick<import('minisearch').Options, 'extractField' | 'tokenize' | 'processTerm'>}
+           */
+          options: {
+            // Configure how fields are extracted from documents
+            extractField: (document, fieldName) => {
+              // Extract frontmatter metadata for search
+              if (fieldName === 'categories' && document.frontmatter?.categories) {
+                return Array.isArray(document.frontmatter.categories)
+                    ? document.frontmatter.categories.join(' ')
+                    : document.frontmatter.categories;
+              }
```

Comment:
I'll consider moving this to a separate helper function with just a parameter, which prop you want to take, for example, `getFrontMatter('categories')`. This whole check will move to this function based on the prop. This will also help in the long run to maintain it if there are some changes to the structure.